Makefile try to compile i3-config-wizard before i3-config-parser.stamp on many threads building

<!--
PLEASE HELP US PROCESS GITHUB ISSUES FASTER BY PROVIDING THE FOLLOWING INFORMATION.
-->

## I'm submitting a…
<!-- Please check one of the following options with "x" -->
<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
<!--
Describe the current behavior,
e.g., »When pressing Alt+j (focus left), the window above the current window is focused.«
-->
A failure occurred in build
```
 GEN      man/i3-config-wizard.xml
  GEN      man/i3-migrate-config-to-v4.xml
  GEN      man/i3-sensible-editor.xml
  GEN      man/i3-sensible-pager.xml
  GEN      man/i3-sensible-terminal.xml
  GEN      man/i3-dump-log.xml
  GEN      man/i3-dmenu-desktop.1
  GEN      man/i3-save-tree.1
  GEN      docs/debugging.html
  GEN      docs/lib-i3test.html
  GEN      docs/lib-i3test-test.html
  GEN      docs/hacking-howto.html
  GEN      docs/userguide.html
../../i3/i3-config-wizard/main.c:109:10: fatal error: GENERATED_config_enums.h: No such file or directory
 #include "GENERATED_config_enums.h"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:2774: i3-config-wizard/i3_config_wizard-main.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/build/i3-gaps-next-git/src/i3/build'
make: *** [Makefile:1332: all] Error 2
 
```
When building on many threads concurrently, the generated Makefile may try to compile i3-config-wizard before i3-config-parser.stamp and result an error where GCC cannot find the to-be-generated header file. This error happens at a low possibility
## Expected Behavior
<!--
Describe the desired behavior you expect after mitigation of the issue,
e.g., »The window left next to the current window should be focused.«
-->
Successfully make without error.
## Reproduction Instructions
<!--
Please provide detailed instructions on how the bug can be reproduced.
E.g., »Open three windows in a V[A H[B C]] layout on a new workspace«
-->
```
autoreconf --force --install
../configure  --prefix=/usr  --sysconfdir=/etc  --disable-sanitizers 
make CPPFLAGS+="-U_FORTIFY_SOURCE" -j$(nproc)
```
In my environment $(nproc) is 24.
## Environment
<!--
Please include your exact i3 version.
Note that we only support the latest major release and the current development version. If you are using an older version of i3, please first update to the current release version and reproduce the issue there.
-->
Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: Binary i3 version:  4.16.1-152-g767a6a52 (2019-03-28, branch "gaps-next") © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16.1-152-g767a6a52 (2019-03-28, branch "gaps-next") (pid 544)
Loaded i3 config: /home/marvelous/.config/i3/config (Last modified: Fri 29 Mar 2019 04:25:31 PM HKT, 8376 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3

</pre>

<!--
Please include your (complete) i3 config with which the issue occurs. You can either paste the file directly or provide a link to a service such as pastebin.

If you would like to help debugging the issue, please try to reduce the config such that it is as close to the default config as possible while still reproducing the issue. This can help us bisect the root cause.
-->
<pre>
</pre>

<!--
Providing a logfile can help us trace the root cause of an issue much quicker. You can learn how to generate the logfile here:
https://i3wm.org/docs/debugging.html

Providing the logfile is optional.
-->
<pre>
Logfile URL:
</pre>

<!--
Please also answer the questions below to help us process your issue faster. If you have any other information to share, please add it here as well.
-->
<pre>
- Linux Distribution & Version: Arch Linux
- Are you using a compositor (e.g., xcompmgr or compton):
</pre>

I try to build f5c0f706480121d1ed501f0af5b84902403ff5d9 , got this:
```
mv -f i3bar/src/.deps/i3bar-parse_json_header.Tpo i3bar/src/.deps/i3bar-parse_json_header.Po
../../i3/docs/i3-pod2html ../../i3/testcases/lib/i3test/Test.pm docs/lib-i3test-test.html
/usr/bin/asciidoc -a toc -n -o docs/hacking-howto.html ../../i3/docs/hacking-howto
../../i3/i3-config-wizard/main.c:109:10: fatal error: GENERATED_config_enums.h: No such file or directory
  109 | #include "GENERATED_config_enums.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
/usr/bin/asciidoc -a toc -n -o docs/userguide.html ../../i3/docs/userguide
make[1]: *** [Makefile:2796: i3-config-wizard/i3_config_wizard-main.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mv -f i3-dump-log/.deps/i3_dump_log-main.Tpo i3-dump-log/.deps/i3_dump_log-main.Po
mv -f i3-input/.deps/i3_input-main.Tpo i3-input/.deps/i3_input-main.Po
mv -f i3-msg/.deps/i3_msg-main.Tpo i3-msg/.deps/i3_msg-main.Po
mv -f i3-nagbar/.deps/i3_nagbar-main.Tpo i3-nagbar/.deps/i3_nagbar-main.Po
mv -f i3bar/src/.deps/i3bar-xcb.Tpo i3bar/src/.deps/i3bar-xcb.Po
make[1]: Leaving directory '/build/i3-gaps-next-git/src/i3/build'
make: *** [Makefile:1337: all] Error 2
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -433,6 +433,9 @@ i3_config_wizard_i3_config_wizard_SOURCES = \
 	i3-config-wizard/main.c \
 	i3-config-wizard/xcb.h
 
+i3_config_wizard_i3_config_wizard_DEPENDENCIES = \
+	$(config_parser_SOURCES)
+
 test_inject_randr15_CPPFLAGS = \
 	$(AM_CPPFLAGS)
 

